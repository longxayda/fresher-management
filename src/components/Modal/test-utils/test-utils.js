import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import manageFRReducer from "../../../redux/fresherManageSlice/fresherManagementSlice";

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: { manageFR: manageFRReducer },
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
