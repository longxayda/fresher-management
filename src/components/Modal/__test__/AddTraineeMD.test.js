import React from "react";
import { screen, render, cleanup, fireEvent } from "@testing-library/react";
import { renderWithProviders } from "../test-utils/test-utils";
import AddTraineeMD from "../AddTraineeModal";

afterEach(cleanup);

describe("Test 1 Add Trainee Modal", () => {
  it("should render Add Button", () => {
    const {} = renderWithProviders(<AddTraineeMD />);
    expect(screen.getByText("Add")).toBeVisible();
  });
  it("shoud render show Modal Add Fresher", () => {
    const {} = renderWithProviders(<AddTraineeMD />);
    fireEvent.click(screen.getByText("Add"));
    expect(screen.getByText("Add New Fresher")).toBeVisible();
  });
});
describe("Test 2 Add Trainee Modal", () => {
  it("should input data firstname", () => {
    const {} = renderWithProviders(<AddTraineeMD />);
    fireEvent.click(screen.getByText("Add"));
    fireEvent.change(screen.getByLabelText("First Name"), {
      target: { value: "Nguyen" },
    });
    expect(screen.getByLabelText("First Name").value).toBe("Nguyen");
  });
  it("should input data lastname", () => {
    const {} = renderWithProviders(<AddTraineeMD />);
    fireEvent.click(screen.getByText("Add"));
    fireEvent.change(screen.getByLabelText("Last Name"), {
      target: { value: "Nam" },
    });
    expect(screen.getByLabelText("Last Name").value).toBe("Nam");
  });
  it("should input data phone", () => {
    const {} = renderWithProviders(<AddTraineeMD />);
    fireEvent.click(screen.getByText("Add"));
    fireEvent.change(screen.getByLabelText("Phone"), {
      target: { value: "0987643244" },
    });
    expect(screen.getByLabelText("Phone").value).toBe("0987643244");
  });
  it("should input data email", () => {
    const {} = renderWithProviders(<AddTraineeMD />);
    fireEvent.click(screen.getByText("Add"));
    fireEvent.change(screen.getByLabelText("Email"), {
      target: { value: "hoangnamqt.2001@gmail.com" },
    });
    expect(screen.getByLabelText("Email").value).toBe(
      "hoangnamqt.2001@gmail.com"
    );
  });
  it("should input data University", () => {
    const {} = renderWithProviders(<AddTraineeMD />);
    fireEvent.click(screen.getByText("Add"));
    fireEvent.change(screen.getByLabelText("University"), {
      target: { value: "HUST" },
    });
    expect(screen.getByLabelText("University").value).toBe("HUST");
  });
  it("should input data Faculty", () => {
    const {} = renderWithProviders(<AddTraineeMD />);
    fireEvent.click(screen.getByText("Add"));
    fireEvent.change(screen.getByLabelText("Faculty"), {
      target: { value: "CNTT" },
    });
    expect(screen.getByLabelText("Faculty").value).toBe("CNTT");
  });
  it("should input data English Level", () => {
    const {} = renderWithProviders(<AddTraineeMD />);
    fireEvent.click(screen.getByText("Add"));
    fireEvent.change(screen.getByLabelText("English Level"), {
      target: { value: "A1" },
    });
    expect(screen.getByLabelText("English Level").value).toBe("A1");
  });
  it("should input data Major", () => {
    const {} = renderWithProviders(<AddTraineeMD />);
    fireEvent.click(screen.getByText("Add"));
    fireEvent.change(screen.getByLabelText("Major"), {
      target: { value: "CNTT" },
    });
    expect(screen.getByLabelText("Major").value).toBe("CNTT");
  });
  it("should input data Graduate Year", () => {
    const {} = renderWithProviders(<AddTraineeMD />);
    fireEvent.click(screen.getByText("Add"));
    fireEvent.change(screen.getByLabelText("Graduate Year"), {
      target: { value: "2020" },
    });
    expect(screen.getByLabelText("Graduate Year").value).toBe("2020");
  });
  it("should input data Headquaters", () => {
    const {} = renderWithProviders(<AddTraineeMD />);
    fireEvent.click(screen.getByText("Add"));
    fireEvent.change(screen.getByLabelText("Headquaters"), {
      target: { value: "Hanoi" },
    });
    expect(screen.getByLabelText("Headquaters").value).toBe("Hanoi");
  });
  it("should have field Select Class", () => {
    const {} = renderWithProviders(<AddTraineeMD />);
    fireEvent.click(screen.getByText("Add"));
    expect(screen.getByLabelText("Class")).toBeVisible();
  });
});
describe("Test 3 Add Trainee Modal", () => {
  it("should have snapshot", () => {
    const { getByText, asFragment, debug } = renderWithProviders(
      <AddTraineeMD />
    );
    const button = getByText("Add");
    fireEvent.click(button);
    const addTraineeModal = asFragment();
    expect(addTraineeModal).toMatchSnapshot();
  });
});
