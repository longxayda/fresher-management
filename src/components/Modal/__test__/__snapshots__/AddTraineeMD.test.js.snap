// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 3 Add Trainee Modal should have snapshot 1`] = `
<DocumentFragment>
  <div
    class="rna-container"
  >
    <div />
  </div>
  <button
    class="btn btn-fill w-100 btn btn-primary"
    type="button"
  >
    <i
      class="fas fa-plus fs-5 fw-bold ms-2 me-2"
    />
    Add
  </button>
</DocumentFragment>
`;
